### PR TITLE
[tempo-distributed] Fix zone aware replication upgrade and rollout operator 

### PR DESCRIPTION
**Pull request by KyriosGN0**
@z0rc, please test upgrade againts this PR, i forgot to add an else clause that adds the default zone if zone aware replication is set to false (weird it only happens during an upgrade

### CODE REVIEW

**Comment by z0rc** [comment]
@KyriosGN0 sure I can test the upgrade path, but first regular checks must pass. I tried to verify upgrade now and got relatively same error as in lint-test check:
```
Error: unable to generate manifests: cannot patch "tempo-ingester" with kind StatefulSet:  "" is invalid: patch: Invalid value:
... 
json: cannot unmarshal string into Go struct field PodSpec.spec.template.spec.affinity of type v1.Affinity
```

**Comment by KyriosGN0** [comment]
@z0rc fixed the tests issue

File: charts/tempo-distributed/values.yaml [hunk 1]
```diff
@@ -992,6 +992,8 @@ enterpriseFederationFrontend:
 
 multitenancyEnabled: false
```
Comment (z0rc) [inline charts/tempo-distributed/values.yaml:994]:
```suggestion
# -- Enable rollout-operator. It must be enabled when using Zone Aware Replication.
rollout_operator:
  enabled: false
  
```

File: charts/tempo-distributed/README.md [hunk 2]
```diff
@@ -766,6 +766,7 @@ The memcached default args are removed and should be provided manually. The sett
 | rbac.create | bool | `false` | Specifies whether RBAC manifests should be created |
 | rbac.pspEnabled | bool | `false` | Specifies whether a PodSecurityPolicy should be created |
 | reportingEnabled | bool | `true` | If true, Tempo will report anonymous usage data about the shape of a deployment to Grafana Labs |
+| rollout_operator.enabled | bool | `false` |  |
```
Comment (z0rc) [inline charts/tempo-distributed/README.md:769]: Please regenarate docs after adding description to variable.

File: charts/tempo-distributed/README.md.gotmpl [hunk 3]
```diff
@@ -39,10 +39,10 @@ The command removes all the Kubernetes components associated with the chart and
 
 A major chart version change indicates that there is an incompatible breaking change needing manual actions.
 
-### from Chart versions < 1.11.0
+### from Chart versions < 1.12.0
```
Comment (z0rc) [inline charts/tempo-distributed/README.md.gotmpl:42]: "Less than" sign is confusing here. AFAIU with recent changes it should be 
```suggestion
### from Chart versions < 1.13.1
```
Or 1.13.0 IDK.

**Review by z0rc — CHANGES_REQUESTED** [review]
Upgrade from 1.12.0 works on my end, regular ingester statefulset remains in place.

Minor comments, otherwise LGTM.

File: charts/tempo-distributed/values.yaml [hunk 4]
```diff
@@ -992,6 +992,8 @@ enterpriseFederationFrontend:
...
 
+rollout_operator:
+  enabled: false
```
Comment (z0rc) [inline charts/tempo-distributed/values.yaml:996]:
```suggestion
# -- Enable rollout-operator. It must be enabled when using Zone Aware Replication.
rollout_operator:
  enabled: false
  
```

**Comment by KyriosGN0** [comment]
fixed! @z0rc thank you so much!

File: charts/tempo-distributed/values.yaml [hunk 5]
```diff
@@ -992,6 +992,10 @@ enterpriseFederationFrontend:
 
 multitenancyEnabled: false
 
+# -- Enable rollout-operator. It must be enabled when using Zone Aware Replication.
+rollout_operator:
+  enabled: false
```
[the remark this answers is not in the document]
Comment (KyriosGN0) [reply charts/tempo-distributed/values.yaml]: fixed!

**Review by z0rc — APPROVED** [review]

**Comment by KyriosGN0** [comment]
@maorfr @faustodavid @swartz-k @BitProcessor 
Sorry for the ping, can anyone take a look to see if all is good?

**Comment by MikaelElkiaer** [comment]
I wonder if this is related to an issue I am seeing.

I upgraded from `1.12.0` to `1.13.0` and all my ingesters disappeared.
I had to roll back to `1.12.0` to restore my them.

<details><summary>My values are:</summary>
<p>

```yaml
gateway:
  enabled: true
global_overrides:
  defaults:
    metrics_generator:
      processors:
        - service-graphs
        - span-metrics
memberlist:
  # INFO: See https://github.com/grafana/loki/issues/10172#issuecomment-1763475167
  cluster_label: tempo
  join_members:
    - dns+{{ include "tempo.fullname" . }}-gossip-ring.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}:{{ .Values.memberlist.bind_port }}
metaMonitoring:
  serviceMonitor:
    enabled: true
    labels:
      release: prom-stack
metricsGenerator:
  enabled: true
traces:
  otlp:
    http:
      enabled: true
    grpc:
      enabled: true
distributor:
  config:
    log_received_spans:
      enabled: true
```

</p>
</details>

**Comment by KyriosGN0** [comment]
@MikaelElkiaer is it exactly that. this PR aims to solve this issue

**Comment by KyriosGN0** [comment]
@mapno @faustodavid @BitProcessor @swartz-k again sorry for the ping but I would like to get this merged asap

**Review by mapno — APPROVED** [review]
Thanks for the fix

**Comment by KyriosGN0** [comment]
@faustodavid @swartz-k @BitProcessor, missing one more approve in order to fix this 
is there something i could to to help with the review ?

**Review by faustodavid — APPROVED** [review]
Thanks for your contribution!

**Comment by faustodavid** [comment]
Can you please update the branch?

**Comment by KyriosGN0** [comment]
@faustodavid thank you very much. updated branch

**Pull request merged**